### ISSUE TITLE
Missing Form Labels - Create New Generic Work

### ISSUE THREAD

**Issue by mtribone** [samvera-labs/nurax-pre2023#123]
### Descriptive summary

When creating a new generic work and using the WAVE accessibility toolbar, there are several form labels missing for form controls. 

They are:

1. Add Files button
1. Add Folder button
1. Date picker for embargo and lease

### Rationale

If a form control does not have a properly associated text label, the function or purpose of that form control may not be presented to screen reader users. Form labels also provide visible descriptions and larger clickable targets for form controls.

### Expected behavior
All form controls should have labels.

### Actual behavior

Some inputs and buttons are missing labels

### Steps to reproduce the behavior

1. Select Works from the left-hand menu
1. On the works screen select "Add new work"
1. Select "generic Work" from the modal
1. Validate the form with the WAVE toolbar
